invisible_cities/cities/components.py

```diff
@@ -152,6 +154,39 @@ def print_every_loop(target):
     return print_every_loop
 
 
+def collect():
+    """Return a future/sink pair for collecting streams into a list."""
+    def append(l,e):
+        l.append(e)
+        return l
+    return fl.reduce(append, initial=[])()
+
+
+def copy_mc_info(files_in : List[str], h5out: tb.File, event_numbers: List[int]):
+    """Copy to an output file the MC info of a list of selected events.
+
+    Parameters
+    ----------
+    files_in : List of strings
+        Name of the input files.
+    file_out : tables.File
+        The output h5 file.
+    event_numbers : List[int]
+        List of event numbers for which the MC info is copied to the output file.
+    """
+
+    writer = mcinfo_io.mc_info_writer(h5out)
+
+    for f in files_in:
+        with tb.open_file(f, "r") as h5in:
+            try:
+                event_numbers_in_file = h5in.root.MC.extents.cols.evt_number[:]
+                event_numbers_to_copy = set(event_numbers_in_file).intersection(event_numbers)
+                mcinfo_io.copy_mc_info(h5in, writer, list(event_numbers_to_copy))
```

Comment:
```suggestion
                mcinfo_io.copy_mc_info(h5in, writer, sorted(event_numbers_to_copy))
```
How aout this? I am afraid the set returns them in a random order

Reply:
But wait, this is wrong, we want to have them in the same order as in all the other tables, which means they cant be sorted but have the exact order of event_numbers list

Reply:
You are right... I suggest
```
events_numbers_to_copy = tuple(evt for evt in event_numbers if evt in event_numbers_in_file)
```
or some other implementation using `filter`

Reply:
In fact, independently of the implementation, there is still a problem that I documented with a test. If the list `event_numbers` contains event numbers that are present in more than one file, then the output file will contain more events than `len(event_numbers)`. I'm not sure there is a way to fix that easily.

Reply:
I dont think you can solve the duplicated events in one file problem in this PR, so Id say forget about that possibility for now...And the efficiency of lists vs sets is really not important in this function, since the slowest part (by couple of orders of magnitude) is the writer itself

Reply:
Yeah, we cannot deal with that right now